spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java

```diff
@@ -139,28 +148,124 @@ public StreamBuilder streamBuilder(DataFlowOperations dataFlowOperations){
 		return Stream.builder(dataFlowOperations);
 	}
 
-	@ConditionalOnProperty(prefix = DataFlowPropertyKeys.PREFIX + "client.authentication", name = "client-id")
 	@Configuration
-	static class ClientCredentialsConfiguration {
+	@ConditionalOnProperty(prefix = DataFlowPropertyKeys.PREFIX + "client.authentication", name = "client-id")
+	@EnableConfigurationProperties(DataFlowClientProperties.class)
+	static class DataFlowClientConfiguration {
+
+		private static Log logger = LogFactory.getLog(DataFlowClientConfiguration.class);
+
+		private static final String DEFAULT_REGISTRATION_ID = "default";
+
+		@Bean
+		public DataFlowOperations dataFlowOperations(@Nullable RestTemplate restTemplate,
```

Comment:
We should not create template here as it's already done above in its own method and we also need ConditionalOnMissingBean as it's done there.